log4j2/src/main/java/com/google/common/flogger/backend/log4j2/Log4j2SimpleLogEvent.java

```diff
@@ -125,6 +142,26 @@ private StackTraceElement getLocationInfo() {
         logSite.getLineNumber());
   }
 
+  private StringMap createContextMap(ContextDataProvider contextDataProvider) {
+    StringMap contextData = ContextDataFactory.createContextData(logData.getMetadata().size());
+    MetadataProcessor
+            .forScopeAndLogSite(contextDataProvider.getMetadata(), logData.getMetadata())
+            .process(Log4j2MetadataHandler.getDefaultMetadataHandler(), new Log4j2KeyValueHandler(contextData));
```

Comment:
I think this would work with the method reference:

```
MetadataProcessor
            .forScopeAndLogSite(contextDataProvider.getMetadata(), logData.getMetadata())
            .process(HANDLER, contextData::putValue);
```

Reply:
done